src/sync/incomingSyncUtils.js

```diff
@@ -632,6 +640,27 @@ export const createOrUpdateRecord = (database, settings, recordType, record) =>
       numberSequence.addNumberToReuse(numberToReuse);
       break;
     }
+    case 'Pref': {
+      const { item } = record;
+      if (item === 'store_preferences') {
+        try {
+          const recordData = JSON.parse(record?.data);
```

Comment:
no need for optional chaining here

Reply:
Yeah, suppose not as the `catch` would catch the `TypeError`.  b9eb872f1876e6978720fb0b4b5956fc3c5b1ff9.

Reply:
I meant record?.data would check that record is undefined right? (at this stage record has to be defined)